### PR TITLE
resolve tasks when IntelliSense is disabled

### PR DESCRIPTION
**Pull request by elahehrashedi**
bugfix: https://github.com/microsoft/vscode-cpptools/issues/9446

### CODE REVIEW

File: Extension/src/Debugger/configurationProvider.ts [hunk 1]
```diff
@@ -148,6 +147,18 @@ export class DebugConfigurationProvider implements vscode.DebugConfigurationProv
 
         if (config.preLaunchTask) {
             config.configSource = this.getDebugConfigSource(config, folder);
+            const isIntelliSenseDisabled: boolean = new CppSettings((vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0) ? vscode.workspace.workspaceFolders[0]?.uri : undefined).intelliSenseEngine === "Disabled";
+            // Run the build task if IntelliSense is disabled.
+            if (isIntelliSenseDisabled) {
+                try {
+                    await cppBuildTaskProvider.runBuildTask(config.preLaunchTask);
```
Comment (bobbrow) [inline Extension/src/Debugger/configurationProvider.ts:154]: I think we should check if `config.preLaunchTask` has a value first.  That's what `resolvePreLaunchTask` does.

Comment (bobbrow) [reply Extension/src/Debugger/configurationProvider.ts:154]: Nevermind. It's on line 148.

**Comment by sean-mcmanus** [comment]
Should we get this into 1.10.6?

**Review by sean-mcmanus — APPROVED** [review]

**Comment by sean-mcmanus** [comment]
> Should we get this into 1.10.6?

It seems like we should.

**Comment by bobbrow** [comment]
> Should we get this into 1.10.6?

If it's not too much trouble, this seems like a pretty safe change.

**Review by sean-mcmanus — APPROVED** [review]

**Pull request merged**